src/DynamoCoreWpf/Controls/StartPage.xaml.cs

```diff
@@ -367,23 +374,41 @@ private void RefreshBackupFileList(IEnumerable<string> filePaths)
             IEnumerable<string> filePaths)
         {
             files.Clear();
+
             foreach (var filePath in filePaths.Where(x => x != null))
             {
                 try
                 {
+                    // Skip files which were moved or deleted (consistent with Revit behavior)
+                    if (!System.IO.File.Exists(filePath)) continue;
+
                     var extension = Path.GetExtension(filePath).ToUpper();
                     // If not extension specified and code reach here, this means this is still a valid file
                     // only without file type. Otherwise, simply take extension substring skipping the 'dot'.
                     var subScript = extension.IndexOf(".") == 0 ? extension.Substring(1) : "";
                     var caption = Path.GetFileNameWithoutExtension(filePath);
 
+                    // Measure the performance for this region
+                    // deserializes the file only once
+                    var jsonObject = DeserializeJsonFile(filePath); 
```

Comment:
I was testing this fork and found an issue here, you are only considering files saved after Dynamo 2.0 (Json files). This code will crash if any of the recent files is an XML dyn file, although less likely these days. Please refer to the file opening logic to check if the file is a valid Json file first before deserializing it as a json object. For XML files, we probably will miss those fields and they can be empty for now. @Amoursol @Jingyi-Wen What do you think?

Reply:
This was taken off-line and discussed within the team. Now added a custom render displayed inside the 'Author' field in Grid view.